core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -1172,7 +1173,40 @@ public PhysicalOperation visitPatternRecognition(PatternRecognitionNode node, Lo
                 nextOutputChannel++;
             }
 
-            // TODO here go window functions in the following channels, similarly to measures
+            // process window functions
+            ImmutableList.Builder<WindowFunctionDefinition> windowFunctionsBuilder = ImmutableList.builder();
+            for (Map.Entry<Symbol, WindowNode.Function> entry : node.getWindowFunctions().entrySet()) {
+                // window functions outputs go in remaining channels starting after the last measure channel
+                outputMappings.put(entry.getKey(), nextOutputChannel);
+                nextOutputChannel++;
+
+                // the common base frame applies to all window functions. This frameInfo will be ignored.
+                FrameInfo ignoredFrame = new FrameInfo(ROWS, CURRENT_ROW, Optional.empty(), Optional.empty(), UNBOUNDED_FOLLOWING, Optional.empty(), Optional.empty(), Optional.empty(), Optional.empty());
```

Comment:
Having to construct and pass an artificial FrameInfo that will be ignored is a bit of a code smell. We should consider making it optional for the call that needs it.